.github/workflows/test-build.yml

```diff
@@ -61,10 +61,60 @@ jobs:
         env:
           MODULE: ${{ matrix.module }}
 
+  build-base-image:
+    name: Build base image
+    runs-on: ubuntu-latest
+    needs: [test]
+
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Cache Docker layer
```

Comment:
Can we use either the [Cache Backend API](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#cache-backend-api) or [Registry Cache](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#registry-cache) here instead?